src/react-components/code-sample.state.tsx

```diff
@@ -43,21 +43,31 @@ type Action =
   | { type: "FETCH_INIT" }
   | { type: "FETCH_SUCCESS"; payload: FetchSuccessPayload }
   | { type: "FETCH_FAILURE"; payload: Error }
-  | { type: "SET_LANGUAGE"; payload: string };
+  | { type: "SELECT"; payload: SafeGetSnippetParams };
 
-function safeGetSnippetForLanguage(
+type SafeGetSnippetParams = {
+  operationId?: string | undefined;
```

Comment:
operationId may not exist - probably better to use method+path. that always exists.